### PR TITLE
optimize mysql undo executor

### PR DESCRIPTION
**Pull request by xuzhiyiYoh**
Optimize undo sql builder and easier to read.

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/undo/mysql/MySQLUndoInsertExecutor.java [hunk 1]
```diff
@@ -50,17 +56,8 @@ protected String buildUndoSQL() {
...
-        boolean first = true;
-        for (Field field : row.getFields()) {
-            if (field.getKeyType() == KeyType.PrimaryKey) {
-                where.append(keywordChecker.checkAndReplace(field.getName()) + " = ?");
-            }
-
-        }
-        return mainSQL.append(where).toString();
+        Field pkField = row.primaryKeys().get(0);
+        return String.format(DELETE_SQL_TEMPLATE, keywordChecker.checkAndReplace(sqlUndoLog.getTableName()), pkField.getName());
```
Comment (xingfudeshi) [inline rm-datasource/src/main/java/io/seata/rm/datasource/undo/mysql/MySQLUndoInsertExecutor.java]: pkField.getName()->keywordChecker.checkAndReplace(pkField.getName()

**Comment by xuzhiyiYoh** [comment]
Thanks for your review, I have fixed it and enhanced the ut. 
By the way, why there are two different field rows added into before image(see the ut method `testDeleteKeywordCheck`).
@xingfudeshi 


image

**Comment by xingfudeshi** [comment]
> Thanks for your review, I have fixed it and enhanced the ut.
> By the way, why there are two different field rows added into before image(see the ut method `testDeleteKeywordCheck`).
> @xingfudeshi
> 
> image

The field name in the second row of data actually has no effect on this UT.it should be a mistake.

File: rm-datasource/src/main/java/io/seata/rm/datasource/undo/mysql/MySQLUndoUpdateExecutor.java [hunk 2]
```diff
@@ -44,24 +54,13 @@ protected String buildUndoSQL() {
...
-
-        }
-        return mainSQL.append(where).toString();
+        Field pkField = row.primaryKeys().get(0);
+        List<Field> nonPkFields = row.nonPrimaryKeys();
+        String updateColumns = nonPkFields.stream()
+            .map(field -> keywordChecker.checkAndReplace(field.getName()) + " = ?")
+            .collect(Collectors.joining(", "));
+        return String.format(UPDATE_SQL_TEMPLATE, keywordChecker.checkAndReplace(sqlUndoLog.getTableName()),
+                             updateColumns, keywordChecker.checkAndReplace(pkField.getName()));
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/undo/mysql/MySQLUndoUpdateExecutor.java]: tableName->keywordChecker.checkAndReplace

**Comment by a364176773** [comment]
由于前期Seata社区治理规范问题部分代码作者未签署CLA，可能引发社区知识产权风险问题。请所有在Seata社区贡献过代码（包含：主项目、官网、samples和多语言项目等）的 contributor 帮忙在这个链接登录github账号签署相应的开发者CLA，https://cla-assistant.io/seata/seata  。2023.1.31 未签署CLA的代码将会被rewrite，拜托大家帮忙签一下。

**Pull request merged**